annotations/text_upload.py

```diff
@@ -0,0 +1,94 @@
+
+from annotations.models import Text, TextCollection, RelationSet
+from repository.models import Repository
+from repository.managers import RepositoryManager
+from django.shortcuts import get_object_or_404, render
+from django.http import HttpResponse
+
+
+
+
+
+def repository_text_content(user, repository_id, text_id, content_id, part_of_id, action, project_id):
+    repository = get_object_or_404(Repository, pk=repository_id)
+
+    manager = RepositoryManager(repository.configuration, user=user)
+    # content_resources = {o['id']: o for o in resource['content']}
+    # content = content_resources.get(int(content_id))    # Not a dict.
```

Comment:
?

Reply:
I didn't write this. This started life as a view which basically does everything I need it to do so I repurposed it into a helper function. I did not spend time to throughly figure out how it works it magic but i use it to add the text to the users repo if it does not exist already. I am not sure why that it commented out but thats the way I found it. Ill remove the extra lines.